brake/backends/cachebe.py

```diff
@@ -51,12 +52,13 @@ def _keys(self, func_name, request, ip=True, field=None, period=None):
 
     def count(self, func_name, request, ip=True, field=None, period=60):
         """Increment counters for all relevant cache_keys given a request."""
-        counters = dict((key, 1) for key in self._keys(
+        counters = dict((key, (1, time.time() + period)) for key in self._keys(
             func_name, request, ip, field, period))
         counters.update(cache.get_many(counters.keys()))
-        for key in counters:
-            counters[key] += 1
-        cache.set_many(counters, timeout=period)
+        for key, value in counters.items():
+            count, expiration = value
+            count += 1
+            cache.set(key, (count, expiration), timeout=expiration - time.time())
```

Comment:
Ah, I see. It's hard to follow what's happening here, because it's assigned implicitly on line 55. 

Can you add some tests that make sure that these conditions are invariant?

I'm especially wondering what happens on the expiration boundary. Say the distance between your expiration and time.time in line 61 is greater than the period duration? We'd have a negative timeout.

Although, it looks like memcached would just set the expiration in the recent past (and it would therefore be expired): https://github.com/memcached/memcached/blob/master/memcached.c#L169

Seriously, though, this logic change needs testing.